DataSource/Util/ChoicesBuilder.php

```diff
@@ -22,39 +22,72 @@ public function __construct(EntityManager $entityManager)
     /**
      * @param array|callable $choicesConfiguration
      *
-     * @return array|null
+     * @return array
      *
      * @throws \Exception
      */
     public function build($choicesConfiguration)
     {
-        $choices = null;
-
         if (
             is_array($choicesConfiguration) &&
```

Comment:
The thing is that if something `is_array`, it could also be `is_callable`, so there is a bit tricky logic here. That's why I didn't want to change this logic and I wanted to keep previous behaviour as less modified as possible.